torch_em/data/segmentation_dataset.py

```diff
@@ -127,10 +139,11 @@ def _get_sample(self, index):
                 if sample_id > self.max_sampling_attempts:
                     raise RuntimeError(f"Could not sample a valid batch in {self.max_sampling_attempts} attempts")
 
-        # squeeze the singleton spatial axis if we have a spatial shape that is larger by one than self._ndim
-        if len(self.patch_shape) == self._ndim + 1:
-            raw = raw.squeeze(1 if self._with_channels else 0)
-            labels = labels.squeeze(1 if self._with_label_channels else 0)
+        if self.patch_shape is not None:
```

Comment:
You can combine this into a single condition: `if self.patch_shape is not None and ...`